src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/reporting/BatchUtils.kt

```diff
@@ -48,3 +53,23 @@ class HeaderWriter(private val header: String) : FlatFileHeaderCallback {
     writer.write(header)
   }
 }
+
+class LoggingWriter<T> : ItemWriter<T> {
+  companion object : KLogging()
+
+  override fun write(items: MutableList<out T>) {
+    logger.info(items.toString())
+  }
+}
+
+interface SentReferralProcessor<T> : ItemProcessor<Referral, T> {
+  companion object : KLogging()
+
+  fun processSentReferral(referral: Referral): T?
+
+  override fun process(referral: Referral): T? {
+    logger.debug("processing referral {}", StructuredArguments.kv("referralId", referral.id))
+    if (referral.sentAt == null) throw RuntimeException("invalid referral passed to report processor; referral has not been sent")
```

Comment:
nit: log message still uses 'report processor' - should read 'sent referral processor'